app/components/toolbarSearch/toolbarSearchDirective.js

```diff
@@ -19,12 +20,29 @@
             };
 
             function toolbarSearchLink(scope) {
-
                 var vm = scope;
-                /**
-                 *
-                 */
+
+
                 vm.filter = searchFilter;
+                vm.filterArray = [];
+                vm.suggestedKeywords = [];
+                vm.textSearchInput = {value: '', previousLength: 0 };
```

Comment:
can you explain the usage if this variable as well? Not sure where previousLength comes into play.